lib/puppet/resource/status.rb

```diff
@@ -195,10 +195,14 @@ def initialize_from_hash(data)
         @failed = data['failed']
         @failed_to_restart = data['failed_to_restart']
         @corrective_change = data['corrective_change']
-        @events = data['events'].map do |event|
-          # Older versions contain tags that causes Psych to create instances directly
-          event.is_a?(Puppet::Transaction::Event) ? event : Puppet::Transaction::Event.from_data_hash(event)
-        end
+        @events = if !out_of_sync? && Puppet[:exclude_unchanged_resources]
+                    []
+                  else
+                    data['events'].map do |event|
+                      # Older versions contain tags that causes Psych to create instances directly
+                      event.is_a?(Puppet::Transaction::Event) ? event : Puppet::Transaction::Event.from_data_hash(event)
+                    end
+                  end
```

Comment:
I think we want the conditional logic in the `to_data_hash` method?

Reply:
oh, I think I need to remove this part if we go the route of [this](https://github.com/puppetlabs/puppet/pull/8952/commits/11edf4fbfc42d7b9d83477edb98490c540321455)?